### PR TITLE
541B - Update all endpoints to use CarbonAwareParameters 

### PR DESCRIPTION
**Pull request by bderusha**
Issue Number: 541

## Summary
Update CarbonAwareAggregator methods and all corresponding codepaths to use the CarbonAwareParameters class for managing user input.

## Changes

- Add all required properties to CarbonAwareParameters.
- Add swagger filters to ensure accuracy of OpenAPI spec when controllers don't own all validation.
- Change aggregator method signature for `GetEmissionsDataAsync` and update dependencies / tests.
- Change aggregator method signature for `GetCurrentForecastDataAsync` and update dependencies / tests.
- Change aggregator method signature for `CalculateAverageCarbonIntensityAsync` and update dependencies / tests.
- Change aggregator method signature for `GetForecastDataAsync` and update dependencies / tests. 

## Checklist

- [X] Local Tests Passing?
- [X] CICD and Pipeline Tests Passing?
- [X] Added any new Tests?
- [X] Documentation Updates Made?
- [X] Are there any API Changes? NO
- [X] This is not a breaking change.

### CODE REVIEW

**Review by gfmatthews — APPROVED** [review]
Now with the time to read through this - all makes sense. Nothing nitpicky or major from my end.

File: src/CarbonAware.WebApi/src/Controllers/CarbonAwareController.cs [hunk 1]
```diff
@@ -249,34 +208,28 @@ public async Task<IActionResult> GetAverageCarbonIntensity([FromQuery, BindRequi
...
     [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ValidationProblemDetails))]
     [ProducesResponseType(StatusCodes.Status500InternalServerError, Type = typeof(ValidationProblemDetails))]
     [HttpPost("average-carbon-intensity/batch")]
-    public async IAsyncEnumerable<CarbonIntensityDTO> GetAverageCarbonIntensityBatch(IEnumerable<CarbonIntensityBatchDTO> requestedCarbonIntensities)
+    public async Task<IActionResult> GetAverageCarbonIntensityBatch([FromBody] IEnumerable<CarbonIntensityBatchDTO> requestedCarbonIntensities)
     {
         using (var activity = Activity.StartActivity())
         {
-            foreach (var carbonIntensityBatchDTO in requestedCarbonIntensities)
+            var result = await Task.WhenAll(requestedCarbonIntensities.Select(async carbonIntensityBatchDTO =>
```
Comment (juzuluag) [inline src/CarbonAware.WebApi/src/Controllers/CarbonAwareController.cs]: This is pretty nice to run this way, but I have couple of concerns:
- WattTime simultaneous requests. Seems like this way we can hit the limit pretty easily, running into a WattTime Exception.
- Client waiting for the response.
- Memory growth until all the data is available.
- Input order vs output order (is there any warranty of it)
- Partial failures. How that works

**Review by juzuluag — COMMENTED** [review]
One comment. Otherwise looks good to me.

File: src/CarbonAware.WebApi/src/Controllers/CarbonAwareController.cs (hunk 1 above)
Comment (bderusha) [reply src/CarbonAware.WebApi/src/Controllers/CarbonAwareController.cs]: Good points.  I updated this to use a `foreach` loop again which addresses your concerns around simultaneous requests, ordering, and partial failures.

When we moved the validation logic to be fully owned by the aggregators it really exposed the fact that these yield streaming responses break the expectations of error handling.  This implementation gives us back nice error handling at the expense of client waiting (not a big concern for most use cases, which are programmatic system requests) and memory growth, which is not a huge risk given that the size of large datasets is measured in MB which is not a concern for modern web-scale machines.

These methods are easy enough to refactor when we tackle error handling at a different layer.

**Review by juzuluag — APPROVED** [review]
Thanks for addressing the last comments I had. 👍

**Pull request merged**